### PR TITLE
doc/crimson: document installing crimson with cephadm

### PR DESCRIPTION
**Pull request by rzarzynski**
Signed-off-by: Radosław Zarzyński <rzarzyns@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/dev/crimson/crimson.rst [hunk 1]
```diff
@@ -24,6 +24,45 @@ cloned using git.
 
 .. _ASan: https://github.com/google/sanitizers/wiki/AddressSanitizer
 
+Installing Crimson with ready-to-use images
+===========================================
+
+An alternative to building Crimson from sources is to use container images built
```
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:30]: s/sources/source/

File: doc/dev/crimson/crimson.rst [hunk 2]
```diff
@@ -24,6 +24,45 @@ cloned using git.
...
+
+NOTE: It is known the conveyed procedure is suboptimal but it received both internal
+external quality assurance.::
+
+
+  $ curl -L https://raw.githubusercontent.com/ceph/ceph-ci/wip-bharat-crimson/src/cephadm/cephadm -o cephadm
+  $ cp cephadm /usr/sbin
+  $ vi /usr/sbin/cephadm
+
+In the opened file change ``DEFAULT_IMAGE = 'quay.ceph.io/ceph-ci/ceph:master'``
```
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:42]: s/opened//

**Review by anthonyeleven — CHANGES_REQUESTED** [review]

**Comment by ljx023** [comment]
I tried to deploy crimson using cephadm according to this document and it worked. 

It would be better if adding some description about image build in the future.

**Comment by zdover23** [comment]
jenkins retest this please

**Comment by zdover23** [comment]
@rzarzynski, you're missing the "Signed-off by:" line in your commit.

Here's a link to the instructions for configuring your Ceph environment to append the "Signed-off by:" line:

https://docs.ceph.com/en/quincy/dev/developer_guide/basic-workflow/#configuring-your-local-environment

And here's the section on amending an existing PR:

https://docs.ceph.com/en/quincy/dev/developer_guide/basic-workflow/#amending-your-pr

Note to future me (not to Radosław): It might not be a bad idea to write a step-by-step guide to squashing.

**Comment by rzarzynski** [comment]
@anthonyeleven: Thanks for the review! I addressed the comments. Would you mind to retake a look?

**Comment by zdover23** [comment]
> @anthonyeleven: Thanks for the review! I addressed the comments. Would you mind to retake a look?


@rzarzynski: see https://github.com/ceph/ceph/pull/47082#issuecomment-1195475442

We won't be able to merge this PR until the "Signed-off by:" line is present in the commit message.

**Comment by rzarzynski** [comment]
@zdover23: I think even the original commit has the tag already:

```
commit 2ca9669e96b9e140cd66a522a151bcb0e47b5215
Author: Radoslaw Zarzynski <rzarzynski@redhat.com>
Date:   Wed Jul 13 14:33:06 2022 +0200

    crimson, doc: document installing crimson with cephadm
    
    Signed-off-by: Rados<C5><82>aw Zarzy<C5><84>ski <rzarzyns@redhat.com>
```

However, somehow (hmm, developing on desktop to look on the HTMLized doc?) UTF-8 sneaked into my name and – perhaps – makes the bot angry. Fixing!

**Comment by zdover23** [comment]
> @zdover23: I think even the original commit has the tag already:
> 
> ```
> commit 2ca9669e96b9e140cd66a522a151bcb0e47b5215
> Author: Radoslaw Zarzynski <rzarzynski@redhat.com>
> Date:   Wed Jul 13 14:33:06 2022 +0200
> 
>     crimson, doc: document installing crimson with cephadm
>     
>     Signed-off-by: Rados<C5><82>aw Zarzy<C5><84>ski <rzarzyns@redhat.com>
> ```
> 
> However, somehow (hmm, developing on desktop to look on the HTMLized doc?) UTF-8 sneaked into my name and – perhaps – makes the bot angry. Fixing!

I haven't tested this on this PR myself, but I suspect that the indentation is what's causing the "Signed-off by:" check to fail. Move that line to the left margin and let's run it again.

**Comment by rzarzynski** [comment]
Huh!

```
=================================== FAILURES ===================================
__________________________ TestCommits.test_doc_title __________________________

self = <test_commits.TestCommits object at 0x7f8460fd66a0>

    @pytest.mark.doc_test
    def test_doc_title(self):
        doc_regex = '\nDate:[^\n]+\n\n    doc'
        all_commits = 'git log -z --no-merges origin/%s..%s' % (
            self.target_branch, self.source_branch)
        wrong_commits = list(filterfalse(
            re.compile(doc_regex).search,
            self.command(all_commits).split('\0')))
        if wrong_commits:
            raise AssertionError("\n".join([
                "The title/s of following commit/s is/are not started with 'doc', but they only touch files under 'doc/'. Please make sure the commit titles",
                "are started with 'doc'. See the 'Submitting Patches' guide:",
                "https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst#commit-title",
                ""] +
>               wrong_commits
            ))
E           AssertionError: The title/s of following commit/s is/are not started with 'doc', but they only touch files under 'doc/'. Please make sure the commit titles
E           are started with 'doc'. See the 'Submitting Patches' guide:
E           https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst#commit-title
E           
E           commit f27b1741a40ff0bb056a0df4a8d96b4f472d688a
E           Author: Radosław Zarzyński <rzarzyns@redhat.com>
E           Date:   Wed Jul 13 14:33:06 2022 +0200
E           
E               crimson, doc: document installing crimson with cephadm
E               
E               Signed-off-by: Radoslaw Zarzynski <rzarzyns@redhat.com>
```

**Review by zdover23 — APPROVED** [review]

**Review by anthonyeleven — APPROVED** [review]

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/47283 - Quincy backport

**Pull request merged**